mono/metadata/object.c

```diff
@@ -3119,6 +3119,7 @@ mono_runtime_try_invoke (MonoMethod *method, void *obj, void **params, MonoObjec
 	MONO_REQ_GC_UNSAFE_MODE;
 
 	g_assert (exc != NULL);
+	g_assert (obj || (method->flags & METHOD_ATTRIBUTE_STATIC));
```

Comment:
can this be "if and only if"?
```suggestion
	g_assert ((obj == NULL) == ((method->flags & METHOD_ATTRIBUTE_STATIC) != 0));
```

I dunno.  Too esoteric?

Reply:
I think that's reasonable so long as you're sure there isn't some weird case where that doesn't hold. The main reason my check isn't as strict is because I was solely worried about a non-static method being passed without an object.